sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BulkExecutionThresholds.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.azure.cosmos.implementation.batch.PartitionScopeThresholds;
+import com.azure.cosmos.util.Beta;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Encapsulates internal state used to dynamically determine max micro batch size for bulk operations.
+ * It allows passing this state for one `BulkProcessingOptions` to another in case bulk operations are
+ * expected to have similar characteristics and the context for determining the micro batch size should be preserved.
+ */
+public class BulkExecutionThresholds {
+    private final ConcurrentMap<String, PartitionScopeThresholds> partitionScopeThresholds;
+
+    /**
+     * Constructor
+     */
+    @Beta(value = Beta.SinceVersion.V4_17_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
```

Comment:
Though it doesn't matter, since we are going for GA, but technically this should be `4_18_0`

Reply:
Fixed in next iteration